dicat/lib/dicom_anonymizer_methods.py

```diff
@@ -268,12 +276,30 @@ def pydicom_zapping(dicom_file, dicom_fields):
             try:
                 dicom_dataset.data_element(
                     dicom_fields[name]['Description']).value = new_val
+            except KeyError as ke:
+                # key error are triggered when a key is not found in the DICOM 
+                # file header. If 'forceInsert' is true for this field, it will
+                # be forcefully added here.
+                # 
+                # if force insert tag, check the value change
+                if name in forceInsertTags:
+                    setattr(dicom_dataset, dicom_fields[name]['Description'], new_val)
+                continue
             except:
                 continue
         else:
             try:
                 dicom_dataset.data_element(
                     dicom_fields[name]['Description']).value = ''
+            except KeyError as ke:
+                # key error are triggered when a key is not found in the DICOM 
+                # file header. If 'forceInsert' is true for this field, it will
+                # be forcefully added here.
+                # 
+                # if force insert tag, check the value change
+                if name in forceInsertTags:
+                    setattr(dicom_dataset, dicom_fields[name]['Description'], '')
+                continue
             except:
                 continue
```

Comment:
To avoid duplication of code:

```
        if dicom_fields[name]['Editable'] is True:
            val = new_val
        else:
            val = ''

        try:
            dicom_dataset.data_element(
                dicom_fields[name]['Description']).value = val
        except KeyError as ke:
            # key error are triggered when a key is not found in the DICOM 
            # file header. If 'forceInsert' is true for this field, it will
            # be forcefully added here.
            # 
            # if force insert tag, check the value change
            if name in forceInsertTags:
                setattr(dicom_dataset, dicom_fields[name]['Description'], val)
        except:
            continue
```